### PR TITLE
Add Profile Notes

### LINKED ISSUE #7715: Inbox notes: Complete store details, Edit store details
This note will be a reminder for users to complete the store details (if the store profiler was skipped) or edit the store details if it was already filled out. **This only applies to the segment of users who are part of the task list UI experiment.**

[Skipped profiler]
Title: Add your store details to complete store setup
Content: Complete your store details with important information for setup such as your store's base address.
Segmentation: Users that have skipped the store profiler `wcadmin_storeprofiler_store_details_skip`
Timing: Immediately upon creating a new store
Cadence: once
CTA leads to: Launch store profiler on store details step
CTA label: Add store details

[Completed profiler]
Title: Edit your store details if you need to
Content: Nice work completing your store profile! You can always go back and edit the details you just shared, as needed.
Segmentation: Users that have completed the store profiler
Timing: Immediately upon creating a new store
Cadence: once
CTA leads to: Launch store profiler on store details step
CTA label: Update store details

@pmcpinto I think this just needs the Tracks event for skipping the store profiler. Also, feel free to add anything or edit the copy.

### PR DESCRIPTION
**Pull request by adrianduffell**
Fixes #7715

Adds notes to prompt the store owner to update their store details, or complete it if it was previously skipped.

### Screenshots

Screen Shot 2021-10-29 at 10 26 49 pm
Screen Shot 2021-10-29 at 10 26 44 pm

### Detailed test instructions:

-   On a fresh install, skip the onboarding wizard.
-   Run the `wc_admin_daily` cron task.
-   Join the experiment treatment group using the `treatment - woocommerce_tasklist_progression_headercard_2021_10` bookmarklet.  Note this didn't work for me. An alternative approach is setting the transient option `_transient_abtest_variation_woocommerce_tasklist_progression_headercard_2021_10` value to `treatment`.
-   See that the "Add your store details to complete store setup" note appears.
-   Click "Add store details" on the note.
-   Complete the onboarding wizard.
-   Run the `wc_admin_daily` cron task.
-   See the "Edit your store details if you need to" note.

### CODE REVIEW

**Comment by moon0326** [comment]
Thank you @adrianduffell 👍 

It worked as expected when I manually updated `_transient_abtest_variation_woocommerce_tasklist_progression_headercard_2021_11` option, but I'm not sure if that's the right way to test.

My local storage shows I'm in the `treatment` group, but the initial `_transient_abtest_variation_woocommerce_tasklist_progression_headercard_2021_11` value is set to `control`. I think the initial `_transient_abtest_variation_woocommerce_tasklist_progression_headercard_2021_11` value should be `treatment` as well 🤔

**Comment by adrianduffell** [comment]
Thanks @moon0326, yeah I agree this isn't the best way to test.

As discussed internally, how about we try to solve the reliability issues with testing ExPlat in our next cooldown and evaluate which approach to take before the 2.9 RC?

**Comment by moon0326** [comment]
> Thanks @moon0326, yeah I agree this isn't the best way to test.
> 
> As discussed internally, how about we try to solve the reliability issues with testing ExPlat in our next cooldown and evaluate which approach to take before the 2.9 RC?

As we discussed, let's investigate it during the cool down.

Approving 👍 🚀

**Review by moon0326 — APPROVED** [review]

File: src/API/Notes.php [hunk 1]
```diff
@@ -179,17 +179,55 @@ public function get_items( $request ) {
 
 		$data = array();
 		foreach ( (array) $notes as $note_obj ) {
+			// Hide selected notes for users not in experiment.
+			if ( ! $this->is_tasklist_experiment_assigned_treatment() ) {
```
[the remark this answers is not in the document]
Comment (adrianduffell) [reply src/API/Notes.php]: Good idea, I made this change in 30a2bc1

File: client/inbox-panel/index.js [hunk 2]
```diff
@@ -274,6 +288,22 @@ const InboxPanel = () => {
 
 	const hasNotes = hasValidNotes( notes );
 
+	const momentDate = moment().utc();
+
+	const [
```
[the remark this answers is not in the document]
Comment (adrianduffell) [reply client/inbox-panel/index.js]: Oh yeah nice, we should do this if we switch back to the front-end approach 👍

**Comment by ilyasfoo** [comment]
It seems possible to get both notes displayed at the same time:
image

I think we need to check if onboarding is completed in `wc-admin-complete-store-details` note condition.

I'll do it in a follow-up

**Pull request merged**